asm_to_pil/src/lib.rs

```diff
@@ -76,11 +76,15 @@ pub mod utils {
             .parse::<T>(input)
             .unwrap()
         {
-            ast::parsed::asm::FunctionStatement::Assignment(start, lhs, using_reg, rhs) => {
+            ast::parsed::asm::FunctionStatement::Assignment(start, lhs, reg, rhs) => {
                 AssignmentStatement {
                     start,
-                    lhs,
-                    using_reg,
+                    lhs_with_reg: {
+                        let lhs_len = lhs.len();
+                        lhs.into_iter()
+                            .zip(reg.unwrap_or((0..lhs_len).map(|_| None).collect()))
```

Comment:
```suggestion
                            .zip(reg.unwrap_or(vec![None; lhs_len]))
```
would something like this also work?

Reply:
but actually I don't really understand what this is doing. Why is it using an entire vector of `None`?

Reply:
Yeah think that works!

If an assignment does not specify any list of assignment registers, then all registers are `None`.

E.g.:
```
A, B <== function_with_multi_return(42);
```

Here, we need two assignment registers which are not specified at this point and need to inferred later from the definition of `function_with_multi_return`.

Reply:
Basically `<==` is replaced by `<=_,_,...,_=`

Reply:
This should probably also use the `OutputRegister` enum?

Reply:
This should probably also use the `OutputRegister` enum?